dev-2/Yandex/Yandex/Pages/BasePage.cs

```diff
@@ -0,0 +1,50 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using System;
+
+namespace Yandex.Pages
+{
+    abstract public class BasePage
+    {
+        private const int WAIT_FOR_ELEMENT_TIMEOUT = 20;
+        protected IWebDriver Driver { get; set; }
+
+        protected WebDriverWait Wait { get; set; }
+
+        protected abstract string Url { get; }
+
+        public BasePage(IWebDriver driver)
+        {
+            Driver = driver;
+            Wait = new WebDriverWait(Driver, TimeSpan.FromSeconds(WAIT_FOR_ELEMENT_TIMEOUT));
+            Driver.Manage().Window.Maximize();
+        }
+
+        public void GoTo()
+        {
+            Driver.Navigate().GoToUrl(Url);
+        }
+
+        public void NavigateOnButton(By locator)
+        {
+
+            try
```

Comment:
You don't need try/catch template here. Just find necessary element and make appropriate actions with it